SwiftCompilerSources/Sources/Optimizer/InstructionSimplification/SimplifyBuiltin.swift

```diff
@@ -162,6 +172,23 @@ private extension BuiltinInst {
     uses.replaceAll(with: literal, context)
     context.erase(instruction: self)
   }
+  
+  func optimizeArrayBuiltin(_ context: SimplifyContext) {
+    guard let metatypeInst = operands[0].value as? MetatypeInst,
+          metatypeInst.type.representationOfMetatype(in: parentFunction) == .Thick else {
+      return
+    }
+    
+    // Must be single use so that we can use replaceAll
+    guard metatypeInst.uses.isSingleUse else {
```

Comment:
This condition is not required if you just do `operands[0].set(to: metatype, context)` instead of `replaceAll`. This will leave other uses of the original metatype instruction untouched.

Reply:
Fixed